crates/cli-support/src/js/mod.rs

```diff
@@ -644,6 +648,59 @@ impl<'a> Context<'a> {
             ));
         }
 
+        // If the class is inspectable, generate `toJSON` and `toString`
+        // to expose all readable properties of the class. Otherwise,
+        // the class shows only the "ptr" property when logged or serialized
+        if class.is_inspectable {
+            // Creates a JavaScript object of all readable properties
+            // This looks like { a: this.a, b: this.b }
+            let readable_properties_js_object = format!(
+                "{{{}}}",
+                class
+                    .readable_properties
+                    .iter()
+                    .fold(String::from("\n"), |fields, field_name| {
+                        format!("{}{name}: this.{name},\n", fields, name = field_name)
+                    })
+            );
+
+            dst.push_str(&format!(
+                "
+                toJSON() {{
+                    return {readable_properties};
+                }}
+
+                toString() {{
+                    return JSON.stringify({readable_properties});
```

Comment:
Could this perhaps be `JSON.stringify(this.toJSON())` to avoid duplication?

Reply:
I implemented it that way originally, but I thought it may be confusing if a struct providing its own `toJSON` had the side effect of changing `toString` and `[util.custom.inspect]`. If it’s documented then it shouldn’t be an issue though. I’ll take your advice and mention it in the guide!

Reply:
Yeah sounds good to me!